core/sizes.cpp

```diff
@@ -189,7 +189,7 @@ Sizes merge(const Sizes &a, const Sizes &b) {
 }
```

Comment:
Having not looked at the code changes yet, I think dropping everything is the right behaviour.
It just makes things simpler (also conceptually).